contrib/babelfishpg_common/src/collation.c

```diff
@@ -872,7 +872,7 @@ lookup_collation_table(Oid coll_oid)
 	{
 		coll_info invalid;
 		invalid.oid = InvalidOid;
-		invalid.enc = PG_UTF8;
+		invalid.enc = coll_infos[server_collation_collidx].enc;
```

Comment:
check the validity of `server_collation_collidx` before using it. If server_collation_collidx == NOT_FOUND then we should raise an error or try to initialise it. Or may be, use `get_server_collation_collidx`

Reply:
Modified the Function to raise an error if `get_server_collation_collidx()` returns NOT_FOUND(-1) collidx else use that collidx to get the respective encoding.